### PR TITLE
feat: [CODE-550]: trigger webhook harness scm

### PR DESCRIPTION
**Pull request by abhinav-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49868)

### CODE REVIEW

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

**Comment by abhinav-harness** [comment]
trigger uts

**Review by vcalasansh — APPROVED** [review]
> Reviewed 6 of 13 files at r1, 4 of 5 files at r3, 4 of 6 files at r5, 1 of 1 files at r6, 1 of 1 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49868)* status: :shipit: complete! ✓

**Comment by abhinav-harness** [comment]
trigger ut9

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java [hunk 1]
```diff
@@ -223,6 +225,24 @@ public NGTriggerEntity create(NGTriggerEntity ngTriggerEntity) {
     }
   }
 
+  private void checkForAccessForHarnessScm(NGTriggerEntity ngTriggerEntity) {
```
Comment (rathodmeetsatish) [inline pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java:228]: Add unit test

File: 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/HarnessSCMWebhookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  private ScmConnector getScmConnector(String repoURL) {
+    return HarnessConnectorDTO.builder()
+        .connectionType(GitConnectionType.REPO)
+        .url(repoURL)
+        .apiAccess(getApiAccess())
+        .build();
+  }
+
+  private HarnessApiAccessDTO getApiAccess() {
+    // Using service principal
```
Comment (rathodmeetsatish) [inline 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/HarnessSCMWebhookServiceImpl.java:108]: add unit tests

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]

File: 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/HarnessSCMWebhookServiceImpl.java (hunk 2 above)
Comment (abhinav-harness) [reply 120-ng-manager/src/main/java/io/harness/ng/webhook/services/impl/HarnessSCMWebhookServiceImpl.java:108]: added in followup

File: pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java (hunk 1 above)
Comment (abhinav-harness) [reply pipeline-service/modules/ng-triggers/src/main/java/io/harness/ngtriggers/service/impl/NGTriggerServiceImpl.java:228]: added in followup

**Comment by abhinav-harness** [comment]
trigger messagecheck

**Review by rathodmeetsatish — APPROVED** [review]

**Review by mohitg0795 — APPROVED** [review]

**Review by adivishy1 — APPROVED** [review]
> Reviewed 6 of 13 files at r1, 4 of 5 files at r3, 4 of 6 files at r5, 1 of 1 files at r6, 1 of 1 files at r7, 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49868)* status: 2. unresolved discussions.

**Pull request merged**